### DISCUSSION TITLE
O que estudar para ser um bom programador? Sem ser linguagem de programação?

### DISCUSSION THREAD

**Discussion by romulosardinha** [frontendbr/forum#1596] [Discussão]
A pergunta já diz tudo.

O que estudar para ser um bom programador? Sem ser linguagem de programação?

Vou dar duas de início aqui:

Algoritmos.
Estrutura de dados.

Ajudem ai.. o que acham que seria uma boa lista de assuntos para estudar e se tornar um bom profissional?

**Comment by felipefialho** [comment]
Ontem mesmo [escrevi sobre soft skills](https://twitter.com/felipefialho_/status/1191684813155160064), elas tendem a ser tão importantes quanto habilidades técnicas durante a carreira.

Coisas como:

- Trabalho em equipe
- Profissionalismo
- Comunicação
- Criatividade
- Etc

Estou pensando em escrever um texto aprofundando nisso. 
Mas enfim, considero essencial.

**Comment by theandersonn** [comment]
Generalizando e falando um pouco fora do âmbito de programação.

- Foco
- Organização
- Equilíbrio mental e corporal

São algumas questões que considero importante refletirmos regularmente. 

Hoje o excesso e facilidade de informação, diversão... trouxe também a falta de foco, concentração que se transformam em serviço 'meia-boca'... colaboradores estressados e exaustos.

Julgo importante absorvermos conteúdo que nos ajudem nesse sentido. 

Recentemente li o livro 'O milagre da manhã' e curti bastante, me ajudou.
https://www.amazon.com.br/milagre-manh%C3%A3-Hal-Elrod/dp/8576849941

**Comment by romulosardinha** [comment]
Nice @felipefialho tinha visto este seu twitte, me despertou isso, no que precisamos em geral... nem só na linguagem, tem a forma de pensar, as softs, a organização que o @theandersonn comentou, isso fez muita diferença na minha vida.

**Comment by cauequeiroz** [comment]
Ótimo tópico! 

- Paradigma (OOP, Funcional)
- Design patterns
- Princípios e boas práticas como SOLID, DRY, KISS
- Testes unitários
- Conceitos gerais de Ciência da Computação
- Inglês

**Comment by talitaazevedo** [comment]
Boa tarde! 
O pessoal da Roketseat está postando um conteúdo bem legal a respeito,  o @willianjusten, tem um post muito bom  sobre isso.

Post:::::::::  [65 dicas para ser um programador melhor](https://willianjusten.com.br/65-dicas-para-ser-um-programador-melhor/)


O que eu uso muito no dia a dia::

- Trello => Criar quadros me ajuda muito a  me organizar (uso uma técnica chamada Kanban).

-  Pomodoro => Me ajuda a focar sem perder a noção do tempo, (pausas são importantes).

- Ifttt | Google Calendar => Vinculei meu google calendar com o ifttt, ele automaticamente me gera cartões para eu tratar das minhas atividades no trello. 

- Evernote | Google Keep => Uso bastante para fazer anotações e adicionar lembretes e tals.

Esses itens me ajudam bastante com produtividade.

**Comment by victorkurauchi** [comment]
- Inglês
- Aprimorar leitura detalhada (até porque passamos muito tempo lendo código também)
- Design patterns
- Desenvolver um equilíbrio em sua vida pessoal/profissional, você precisa estar bem consigo mesmo pra performar em seu melhor nível :)

**Comment by romulosardinha** [comment]
Tbm uso muito o trello e o calendar @tnazevedo ! mas ainda não me acostumei com o pomodoro.

**Comment by wilsonneto-swe** [comment]
- Estimativas certeiras (marca td q vc faz para ter base en histórico e não no que vc acha q vai gastar de tempo)
- Arquitetura e Patterns (além do código)
- Gerenciamento de tempo
- Gerenciamento de tarefas (trello / kanban)
- Básico em gerenciamento de projetos com PMBOK (me ajudou demais a organizar meus processos)
- PDCA
- Scrum e XP avançado (não todo o marketing em volta, mas os conceiros bases)
- CI / CD / Deploys / Conteiners / Docker
- Técnicas de produtividade para manter a performance codando do início do dia até o encerramento do ecpediente
- Ferramentas / métodos de trackeamento de tempo (onde vc usa mais tempo? é onde deveria?)
- Ferramentas de trackeamento de bugs ("bugs não devem existir", descubra a raíz e causa deles e os minimize ao máximo, as vezes vc descobre q se mudar um pouco dos processos seus bugs quase somem... O tempo deve ser gasto evoluindo e não corrigindo)

Alguns pontos que me dedico a aprender além de código.

**Comment by iaurg** [comment]
Além das ótimas dicas e soft skills que o pessoal recomendou acima.

O start do [roadmap](https://roadmap.sh/) possui temas úteis para toda a parte de programação:
![image](https://user-images.githubusercontent.com/11761170/68766538-d06eab80-05fd-11ea-967c-e867bcab5349.png)

Tenho um [repositório ](https://github.com/Italox/Computer-Science-Study)(que estou melhorando) para que possa ter acesso à um caminho melhor para entrar no mundo da ciência de computação. Vi que existe muito conteúdo na internet mas pouco foco na base de tudo.

Algo que me ajuda bastante também é o canal do [Akita](https://www.youtube.com/user/AkitaOnRails), ele é um profissional de longa data no mercado e dá umas dicas e formas de pensar muito boas.

De resto aceito melhorias e dicas também.

Até mais!

**Comment by caioincau** [comment]
Comunicação! A maioria dos profissionais que encontrei e que tinham background 100% de tecnologia tinham dificuldade de se comunicar, eu mesmo tinha muita dificuldade no começo da minha carreira e hoje vejo como isso me “atrasou”.

**Comment by eliseumds** [comment]
Inglês, redes, Unix, encoding, datetime e nutrição.
